File: app/src/main/java/com/tomclaw/drawa/dto/Size.java
package com.tomclaw.drawa.dto;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by solkin on 27.02.17.
 */
public class Size implements Parcelable {

    private int width;
    private int height;

    public Size(int width, int height) {
        this.width = width;
        this.height = height;
    }

    private Size(Parcel in) {
        width = in.readInt();
        height = in.readInt();
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(width);
        dest.writeInt(height);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Size> CREATOR = new Creator<Size>() {
        @Override
        public Size createFromParcel(Parcel in) {
            return new Size(in);
        }

        @Override
        public Size[] newArray(int size) {
            return new Size[size];
        }
    };
}

File: app/src/main/java/com/tomclaw/drawa/dto/Size.kt
package com.tomclaw.drawa.dto

import android.os.Parcel
import android.os.Parcelable

class Size(val width: Int,
           val height: Int) : Parcelable {

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeInt(width)
        writeInt(height)
    }

    override fun describeContents(): Int = 0

    companion object CREATOR : Parcelable.Creator<Size> {
        override fun createFromParcel(parcel: Parcel): Size {
            val width = parcel.readInt()
            val height = parcel.readInt()
            return Size(width, height)
        }

        override fun newArray(size: Int): Array<Size?> {
            return arrayOfNulls(size)
        }
    }
}
